core/lib/src/unix.rs

```diff
@@ -0,0 +1,67 @@
+use crate::error::{LaunchError, LaunchErrorKind};
+use std::path::Path;
+use std::fs::{self, File};
+
+#[doc(hidden)]
+pub fn lock_socket<P: AsRef<Path>>(path: P) -> Result<File, LaunchError> {
+    use fs2::FileExt;
+
+    let path = path.as_ref();
+    let lock_path = path.with_extension("lock");
```

Comment:
I'm not sure if this does what we want. If `path` is `foo.bar`, this will convert it to `foo.lock` where as we probably want `foo.bar.lock`.

Reply:
In practice I think I'd name the Unix socket path `foo.sock`/`foo.bar.sock`, so I think it's fine to just replace the extension to `lock`.

Reply:
I'd much prefer to see `foo.sock.lock`.

Reply:
I don't have a strong opinion about this, if you want to change it feel free to do so.